src/cardano_fmt_plug.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * JtR format to crack password protected Cardano 128-byte length legacy secret Keys (a.k.a XPrv).
+ *
+ * This software is Copyright (c) 2022, Pal Dorogi <pal dot dorogi at gmail.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_cardano;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_cardano);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE 1
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "jumbo.h"
+#include "cardano_common.h"
+#include "blake2.h"
+#include "pbkdf2_hmac_sha512.h"
+#include "chacha.h"
+#include "ed25519.h"
+
+#define FORMAT_NAME         "Cardano Encrypted 128-byte Secret Key (a.k.a XPrv)"
+#define FORMAT_LABEL        "cardano"
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20 " SHA512_ALGORITHM_NAME
+#else
+#if ARCH_BITS >= 64
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20  64/" ARCH_BITS_STR
+#else
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20  32/" ARCH_BITS_STR
+#endif
+#endif
+#define BENCHMARK_COMMENT   ""
+#define BENCHMARK_LENGTH    0x107
+#define BINARY_SIZE         0
+#define BINARY_ALIGN        1
+#define SALT_SIZE           sizeof(struct custom_salt)
+#define SALT_ALIGN          sizeof(uint32_t)
+#ifdef SIMD_COEF_64
+#define MIN_KEYS_PER_CRYPT  SSE_GROUP_SZ_SHA512
+#define MAX_KEYS_PER_CRYPT  (SSE_GROUP_SZ_SHA512 * 4)
+#else
+#define MIN_KEYS_PER_CRYPT  1
+#define MAX_KEYS_PER_CRYPT  4
+#endif
```

Comment:
I think 15000 iterations is so high that the `* 4` and a non-SIMD max of 4 are unneeded - can set them to 1 for greater program interactivity at no significant effect on performance.

Reply:
Fixed